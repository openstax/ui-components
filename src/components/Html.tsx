import React from 'react';
import { assertString } from "@openstax/ts-utils/assertions";
import DOMPurify from 'dompurify';

export const Html = (props: React.PropsWithChildren<{block?: boolean; className?: string}>) => {
  if (props.children === undefined) {
    return null;
  }
  const html = DOMPurify.sanitize(assertString(props.children), {ADD_ATTR: ['target']});
  return props.block
    ? <div dangerouslySetInnerHTML={{__html: html}} className={props.className} />
    : <span dangerouslySetInnerHTML={{__html: html}} className={props.className} />
  ;
};
