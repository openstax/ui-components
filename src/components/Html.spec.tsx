import { Html } from "./Html";
import { render } from '@testing-library/react';

describe('Html', () => {
  let root: HTMLElement;

  beforeEach(() => {
    root = document.createElement('main');
    root.id = 'root';
    document.body.append(root);
  });

  it('matches snapshot (Block)', () => {
    render(
      <Html block className="custom-class">
        This is a block of HTML
      </Html>
    );

    expect(document.body).toMatchSnapshot();
  });

  it('matches snapshot (Inline)', () => {
    render(
      <Html className="custom-class">
        This is an inline HTML.
      </Html>
    );

    expect(document.body).toMatchSnapshot();
  });

  it('matches snapshot (Empty)', () => {
    render(<Html />);

    expect(document.body).toMatchSnapshot();
  });
});


