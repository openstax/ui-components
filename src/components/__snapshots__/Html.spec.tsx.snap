// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html matches snapshot (Block) 1`] = `
<body>
  <main
    id="root"
  />
  <div>
    <div
      class="custom-class"
    >
      This is a block of HTML
    </div>
  </div>
</body>
`;

exports[`Html matches snapshot (Empty) 1`] = `
<body>
  <main
    id="root"
  />
  <main
    id="root"
  />
  <main
    id="root"
  />
  <div />
</body>
`;

exports[`Html matches snapshot (Inline) 1`] = `
<body>
  <main
    id="root"
  />
  <main
    id="root"
  />
  <div>
    <span
      class="custom-class"
    >
      This is an inline HTML.
    </span>
  </div>
</body>
`;
